enclave/server.ts

```diff
@@ -51,11 +46,64 @@ const nStates = new ethers.Contract(
     signer
 );
 
+type ClaimedMove = {
+    uFrom: Tile;
+    uTo: Tile;
+};
+
 /*
  * Enclave's internal belief on game state stored in Board object.
  */
 let b: Board;
 
+/*
+ * Bijection between player's public keys and their socket IDs.
+ */
+let idToPubKey = new Map<string, string>();
+let pubKeyToId = new Map<string, string>();
+
+/*
+ * Moves claimed by players. A move is finalized whenever NewMove event is
+ * emitted.
+ */
+let claimedMoves = new Map<string, ClaimedMove>();
+
+/*
+ * Dev function for spawning a player on the map.
+ */
+async function spawn(
+    socket: Socket,
+    l: Location,
+    reqPlayer: Player,
+    sigStr: string
+) {
+    // If player is already spawned in, their socket ID will already be stored
```

Comment:
hmm why do we need this if statement? if it's preventing double spawning, there should be a more direct way to do that

Reply:
it prevents the enclave from doing extra work if a player tries to login more than once on the same connection. it also prevents double spawning. what else can we do here?